cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -5039,6 +5057,154 @@ if(ARROW_S3)
+                                   INTERFACE_INCLUDE_DIRECTORIES
+                                   ${AZURESDK_INCLUDE_DIR})
+  set_property(TARGET Azure::azure-identity
+               PROPERTY INTERFACE_LINK_LIBRARIES OpenSSL::Crypto Azure::azure-core)
+  add_dependencies(Azure::azure-identity azuresdk_ep)
+
+  add_library(Azure::azure-storage-common STATIC IMPORTED)
+  set_target_properties(Azure::azure-storage-common
+                        PROPERTIES IMPORTED_LOCATION ${AZURE_STORAGE_COMMON_STATIC_LIBRARY}
+                                   INTERFACE_INCLUDE_DIRECTORIES
+                                   ${AZURESDK_INCLUDE_DIR})
+  set_property(TARGET Azure::azure-storage-common
+               PROPERTY INTERFACE_LINK_LIBRARIES OpenSSL::Crypto LibXml2::LibXml2 Azure::azure-core)
+  add_dependencies(Azure::azure-storage-common azuresdk_ep)
+
+  add_library(Azure::azure-storage-blobs STATIC IMPORTED)
+  set_target_properties(Azure::azure-storage-blobs
+                        PROPERTIES IMPORTED_LOCATION ${AZURE_STORAGE_BLOBS_STATIC_LIBRARY}
+                                   INTERFACE_INCLUDE_DIRECTORIES
+                                   ${AZURESDK_INCLUDE_DIR})
+  set_property(TARGET Azure::azure-identity
+               PROPERTY INTERFACE_LINK_LIBRARIES Azure::azure-core Azure::azure-storage-common)
+  add_dependencies(Azure::azure-storage-blobs azuresdk_ep)
+
+  add_library(Azure::azure-storage-files-datalake STATIC IMPORTED)
+  set_target_properties(Azure::azure-storage-files-datalake 
+                        PROPERTIES IMPORTED_LOCATION ${AZURE_STORAGE_FILES_DATALAKE_STATIC_LIBRARY}
+                                   INTERFACE_INCLUDE_DIRECTORIES
+                                   ${AZURESDK_INCLUDE_DIR})
+  set_property(TARGET Azure::azure-storage-files-datalake
+               PROPERTY INTERFACE_LINK_LIBRARIES Azure::azure-core Azure::azure-identity Azure::azure-storage-common Azure::azure-storage-blobs)
+  add_dependencies(Azure::azure-storage-files-datalake azuresdk_ep)
+
+  set(AZURE_SDK_VENDORED TRUE)
+  set(AZURESDK_LIBRARIES)
+  list(APPEND
+       AZURESDK_LIBRARIES
+       Azure::azure-core
+       Azure::azure-identity
+       Azure::azure-storage-blobs
+       Azure::azure-storage-common
+       Azure::azure-storage-files-datalake)
+  list(APPEND
+       ARROW_BUNDLED_STATIC_LIBS
+       Azure::azure-core
+       Azure::azure-identity
+       Azure::azure-storage-blobs
+       Azure::azure-storage-common
+       Azure::azure-storage-files-datalake)
+
+  set(AZURESDK_LINK_LIBRARIES ${AZURESDK_LIBRARIES})
+endmacro()
+
+if(ARROW_WITH_AZURE_SDK)
+  if(AZURE_SDK_SOURCE STREQUAL "SYSTEM")
+    resolve_dependency(Azure::azure-core)
+    resolve_dependency(Azure::azure-identity)
+    resolve_dependency(Azure::azure-storage-blobs)
+    resolve_dependency(Azure::azure-storage-common)
+    resolve_dependency(Azure::azure-storage-files-datalake)
```

Comment:
When using system dependencies installed with vcpkg each component is separate. When building with the external project though its all packaged together in one external project.

Reply:
Hmm. This approach doesn't work with `AZURE_SDK_SOURCE=AUTO`.
How about adding `cpp/cmake_modules/FindAzure.cmake` and use `resolve_dependency(Azure)` and using `find_package(Azure::*)`s in `FindAzure.cmake`?

Reply:
I''ll try this. I will need to have a look at some of the existing ones and see if I can work out how to do it.

Reply:
I've had a bit of a look and I'm not sure I understand. I gave it a [try](https://github.com/apache/arrow/pull/36835/commits/414d1c738483c7b48cf234bbaa518460ac5a2779) but what I ended up with seems totally different to other custom `Find<package-name>.cmake`. 

I will have another look at this tomorrow but if you could give me some more guidance that would be appreciated.

Reply:
I've got something now that seems to work and looks about right to me...

Reply:
Marking as resolved because the idea is resolved and there are new comments on my implementation of `FindAzure.cmake`